tests/pipelines/test_pipelines_common.py

```diff
@@ -875,84 +875,93 @@ def test_cfg(self):
+        def callback_inputs_success(pipe, i, t, callback_kwargs):
+            # interate over callback args
+            for tensor_name, tensor_value in callback_kwargs.items():
+                # check that we're only passing in allowed tensor inputs
+                assert tensor_name in pipe._callback_tensor_inputs
+
             return callback_kwargs
 
         inputs = self.get_dummy_inputs(torch_device)
-        inputs["callback_on_step_end"] = callback_inputs_test
+
+        # Test passing in a subset
+        inputs["callback_on_step_end"] = callback_inputs_success
+        inputs["callback_on_step_end_tensor_inputs"] = ["latents"]
+        inputs["output_type"] = "latent"
+        output = pipe(**inputs)[0]
+
+        # Test passing in a everything
+        inputs["callback_on_step_end"] = callback_inputs_success
         inputs["callback_on_step_end_tensor_inputs"] = pipe._callback_tensor_inputs
         inputs["output_type"] = "latent"
+        output = pipe(**inputs)[0]
 
+        def callback_inputs_change_tensor(pipe, i, t, callback_kwargs):
+            is_last = i == (pipe.num_timesteps - 1)
+            if is_last:
+                callback_kwargs["latents"] = torch.zeros_like(callback_kwargs["latents"])
+            return callback_kwargs
+
+        inputs["callback_on_step_end"] = callback_inputs_change_tensor
+        inputs["callback_on_step_end_tensor_inputs"] = pipe._callback_tensor_inputs
+        inputs["output_type"] = "latent"
         output = pipe(**inputs)[0]
         assert output.abs().sum() == 0
 
     def test_callback_cfg(self):
         sig = inspect.signature(self.pipeline_class.__call__)
+        has_callback_tensor_inputs = "callback_on_step_end_tensor_inputs" in sig.parameters
+        has_callback_step_end = "callback_on_step_end" in sig.parameters
 
-        if not ("callback_on_step_end_tensor_inputs" in sig.parameters and "callback_on_step_end" in sig.parameters):
+        if not (has_callback_tensor_inputs and has_callback_step_end):
             return
 
         if "guidance_scale" not in sig.parameters:
             return
 
         components = self.get_dummy_components()
         pipe = self.pipeline_class(**components)
-        pipe = pipe.to(torch_device)
+        pipe.to(torch_device)
         pipe.set_progress_bar_config(disable=None)
+        self.assertTrue(
+            hasattr(pipe, "_callback_tensor_inputs"),
+            f" {self.pipeline_class} should have `_callback_tensor_inputs` that defines a list of tensor variables its callback function can use as inputs",
+        )
 
-        inputs = self.get_dummy_inputs(torch_device)
-        inputs["guidance_scale"] = 1.0
-        out_no_cfg = pipe(**inputs)[0]
-
-        def callback_no_cfg(pipe, i, t, callback_kwargs):
-            if i == 0:
-                for k, w in callback_kwargs.items():
-                    if k in self.callback_cfg_params:
-                        callback_kwargs[k] = callback_kwargs[k].chunk(2)[-1]
```

Comment:
I think I'm a little confused as to which tensor is being chunked and why? If there's no cfg, we wouldn't need to chunk anything right?

Reply:
And I think it makes it a little more challenging from a maintenance perspective to ensure that we always have the right `self.callback_cfg_params` for every single pipeline in order to run the test.